orchagent/cbf/cbfnhgorch.cpp

```diff
@@ -0,0 +1,799 @@
+             */
+            if (nhg.getId() != member->second)
+            {
+                if (!m_members.at(member->first).updateNhAttr())
+                {
+                    SWSS_LOG_ERROR("Failed to update temporary next hop group "
+                                    "member %s of CBF next hop group %s",
+                                    member->first.c_str(), m_key.c_str());
+                    return false;
+                }
+
+                /* If the NHG was promoted, remove it from the temporary NHG map. */
+                if (!nhg.isTemp())
+                {
+                    member = m_temp_nhgs.erase(member);
+                }
+                /*
+                 * If the NHG was just updated, update its current NHG ID value
+                 * in the map.
+                 */
+                else
+                {
+                    member->second = nhg.getId();
+                    ++member;
+                }
+            }
+            else
+            {
+                ++member;
+            }
+        }
+    }
+    /* If the members are different, update the whole members list. */
+    else
+    {
+        /*
+         * Because the INDEX attribute is CREATE_ONLY, we have to remove all
+         * the existing members and sync the new ones back as the order of the
+         * members can change due to some of them being removed, which would
+         * invalidate all the other members following them, or inserting a new
+         * one somewhere in the front of the existing members which would also
+         * invalidate them.
+         */
+        set<string> members_set;
+
+        for (const auto &member : m_members)
+        {
+            members_set.insert(member.first);
+        }
+
+        /* Remove the existing members. */
+        if (!removeMembers(members_set))
+        {
+            SWSS_LOG_ERROR("Failed to remove members of CBF next hop group %s",
+                            m_key.c_str());
+            return false;
+        }
+        m_members.clear();
+        m_temp_nhgs.clear();
+
+        /* Add the new members. */
+        uint8_t index = 0;
+        for (const auto &member : members)
+        {
+            m_members.emplace(member, CbfNhgMember(member, index++));
+        }
+
+        /* Sync the new members. */
+        if (!syncMembers({members.begin(), members.end()}))
+        {
+            SWSS_LOG_ERROR("Failed to sync members of CBF next hop group %s",
+                            m_key.c_str());
+            return false;
+        }
+    }
+
+    /* Update the group map. */
+    if (m_selection_map != selection_map)
+    {
+        sai_attribute_t nhg_attr;
+        nhg_attr.id = SAI_NEXT_HOP_GROUP_ATTR_SELECTION_MAP;
+        nhg_attr.value.oid = gNhgMapOrch->getMapId(selection_map);
+
+        if (nhg_attr.value.oid == SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_WARN("NHG map %s does not exist", selection_map.c_str());
```

Comment:
I've changed to ERROR (technically it can be transient if the map and group are created at the same time but the group gets processed first, but that's unlikely)